lib/psych/visitors/to_ruby.rb

```diff
@@ -383,7 +382,8 @@ def deduplicate key
           if key.is_a?(String)
             # Deduplication is not supported by this implementation,
             # but we emulate it's side effects
-            key.untaint.freeze
+            key.untaint if RUBY_VERSION < '2.7'
```

Comment:
@jeremyevans this `RUBY_VERSION < '2.7'` can be removed because this version of the method is only executed if `String#-@` doesn't exist (meaning MRI 2.2- IIRC)

Reply:
Thanks for the review.  I agree, this can be removed.  Note that the tests don't seem to pass on ruby 2.2 due to the use of String#match? (added in 2.4).

Reply:
Yeah, I think I broke the compatibility in a later PR, I need to check what we did about that. Either we can remove that conditional from the code, or either I need to remove the `match?` optimisation.

Reply:
Answer: we but the minimum version requirement to 2.4: https://github.com/ruby/psych/pull/415

So you can remove the `-@` check.